github/github.go

```diff
@@ -880,6 +880,23 @@ func cloneRequest(r *http.Request) *http.Request {
 	return r2
 }
 
+func formatDuration(d time.Duration) string {
```

Comment:
Also, what will `formatDuration` return on negative durations? Does it handle them properly?

It's possible for a `RateLimitError` to be stored for some time, and its `Error` method can be called after `r.Rate.Reset` is in the past. Then, `r.Rate.Reset.Time.Sub(time.Now())` will be a negative duration, so `formatDuration` needs to be able to handle it.

Perhaps it's a good idea to add a few test cases for `formatDuration`, including a negative duration.